### ISSUE TITLE
SMTP mail not functioning in 3.5.0-rc

### ISSUE THREAD

**Issue by ctbakken** [joomla/joomla-cms#9373]
#### Steps to reproduce the issue
Fill / choose all applicable fields in Mail Settings:
Send Mail -> Yes
Mailer -> SMTP
From email -> my own email address
From Name -> My Name
Disable Mass Mail -> No
SMTP Authentication -> No
SMTP Security -> None / SSL / TLS (tried all)
SMTP Port -> 25 / 587 (tried both)
SMTP Username -> None
SMTP Password -> None
SMTP Host -> My SMTP server, authenticating sender IP-address

Save

Send Test Mail

#### Expected result
Test mail sent

#### Actual result
Error messages:
Test mail could not be sent
Could not instantiate mail function

#### System information (as much as possible)
PHP Built On 	Linux <host name> 3.2.0-98-generic #138-Ubuntu SMP Mon Jan 11 12:33:01 UTC 2016 x86_64
Database Version 	5.5.47-0ubuntu0.12.04.1
Database Collation 	utf8_general_ci
Database Connection Collation 	utf8mb4_general_ci
PHP Version 	5.5.23
Web Server 	Apache
WebServer to PHP Interface 	cgi-fcgi
Joomla! Version 	Joomla! 3.5.0-rc Release Candidate [ Ember ] 10-March-2016 00:30 GMT
Joomla! Platform Version 	Joomla Platform 13.1.0 Stable [ Curiosity ] 24-Apr-2013 00:00 GMT
User Agent 	Mozilla/5.0 (Macintosh; Intel Mac OS X 10.11; rv:44.0) Gecko/20100101 Firefox/44.0 

#### Additional comments
Sendmail not installed on server due to security implications
SMTP working fine on Joomla up to and including 3.4.8 on same server

**Comment by andrepereiradasilva** [comment]
Could be this? https://github.com/PHPMailer/PHPMailer/wiki/Troubleshooting#could-not-instantiate-mail-function

**Comment by ctbakken** [comment]
Probably, but does that mean that J3.5 no longer supports direct sending via SMTP like earlier Joomla versions?
Isn't this a step back securitywise? I've got rid of the problem with rogue scripts injected in my Joomla installations sending thousands of spam emails, by removing the local mail server software from my servers...
In your link I read about isSMTP(), is this something I can install / configure server-wide, or do I have to alter the PHPMailer script in all new Joomla installations?

**Comment by andrepereiradasilva** [comment]
> Probably, but does that mean that J3.5 no longer supports direct sending via SMTP like earlier Joomla versions?

Of course it supports. I have a SMTP working fine in my joomla test server with 3.5.0 RC.

> Isn't this a step back securitywise? I've got rid of the problem with rogue scripts injected in my Joomla installations sending thousands of spam emails, by removing the local mail server software from my servers...

I'm not telling to use sendmail. the link i posted is the oficial phpmailer troubleshoot so you can check what's is going on with your smtp mail server.
the only thing you should need is an smtp mail server that works fine.

> In your link I read about isSMTP(), is this something I can install / configure server-wide, or do I have to alter the PHPMailer script in all new Joomla installations?

isSMTP is a PHPMailer function the link i posted is the oficial phpmailer troubleshoot.
Please never change joomla core files... when you update they will be replaced...

In conclusion, you seem to be hitting a phpmailer error because of your smtp server config, i guess.

also, don't forget that from 3.4.8 to 3.5.0 RC the version of phpmailer lib changed so there can be additional checks.

- 3.4.8: 5.2.9 (https://github.com/joomla/joomla-cms/blob/3.4.x/libraries/vendor/phpmailer/phpmailer/class.phpmailer.php#L34)
- 3.5.0 RC: 5.2.14 https://github.com/joomla/joomla-cms/blob/staging/libraries/vendor/phpmailer/phpmailer/class.phpmailer.php#L34

Changelog: https://github.com/PHPMailer/PHPMailer/blob/master/changelog.md

**Comment by kigh143** [comment]
hey sorry about that,  I made a module that was to send an email notification from the super user to the person who has contacted them. 
if your are sending the mail while on localhost your get that error, please modify these mail settings under the system menu -> select global configurations, then under the server tab modify the contents of your mail settings
turn on send mail
**mailer ** set it to SMTP
then the from email  set it to your email address from either, gmail, live, etc
from name : add a name username
turn on SMTP Authentication 
set SMTP security to TLS
SMTP port 25, or 2525
SMTP username : your email address
SMTP password the password you user for the provided email address
then for the host. user smtp.gmail.com for gmail, smtp.live.com for live.com etc 
hope this will help your 
thanks and regards  enjoy being at joomla.

**Comment by kigh143** [comment]
please also show your system information for your configuration file

**Comment by wilsonge** [comment]
Another report of this on the forum http://forum.joomla.org/viewtopic.php?f=706&t=913705

**Comment by krogias** [comment]
Hi,
I know that this is a RC issue tracker, but I've experienced the same issue after updating to 3.5 stable.
Database Version: 5.6.21, PHP Version: 5.6.3, Web Server: Apache/2.4.10 (Win32) PHP/5.6.3

My mail server is working fine and is updated.
I had to roll back to 3.4.8.. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9373">issues.joomla.org/joomla-cms/9373</a>.</sub>

**Comment by brianteeman** [comment]
@krogias Can you give some more information please. How and where are you sending mail. Is it a core component or an extension. What were the results of using the test email button in global configuration<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9373">issues.joomla.org/joomla-cms/9373</a>.</sub>

**Comment by krogias** [comment]
It was the core component (contact form). Sending as a guest to a contact. 
I have some automated messages to admins from Akeeba Admin Tools, which didn't work either.
I Don't have any other info because I rolled back. 
My settings:
Send Mail -> Yes
Mailer -> SMTP
Disable Mass Mail -> No
SMTP Authentication -> Yes
SMTP Security -> None
SMTP Port -> 25
SMTP Host -> My SMTP server
Everything works fine now in 3.4.8.
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9373">issues.joomla.org/joomla-cms/9373</a>.</sub>

**Comment by wilsonge** [comment]
What hosting provider are you using? SMTP mails are definitely working for some people - so it must be some sort of server specific bug :/

**Comment by krogias** [comment]
As stated above SMTP Host-> My SMTP server, which means that I use my own mail server. I don't think it is a mail server issue, because it's one of the best commercial mail servers.
I Know that Joomla! 3.5 updated the PHPMailer library and I don't think it's right to change our mail server settings whenever Joomla updates it's libraries.

**Comment by wilsonge** [comment]
https://github.com/PHPMailer/PHPMailer/releases PHP Mailer released a "important security fix" as you can see from the changelog. I'm not in a place where I can roll back the PHPMailer upgrade. That's why we need to figure out what the issue is - we have a class in Joomla that we can apply fixes if it comes down to that - but without being able to reliably reproduce the issue we can't fix the issue.

**Comment by krogias** [comment]
That's very bad.

**Comment by kigh143** [comment]
Hello, I was making a module that adds the contact us functionality to a website or blog. U configured my settings to those and it worked.
Sent from my ASUS

-------- Original Message --------

**Comment by krogias** [comment]
@kigh143 Sorry, but I cannot set SMTP Authentication to "No"

**Comment by wilsonge** [comment]
@krogias What it boils down to is without being able to debug or reproduce your issue in any way I can't help you :( If you're not willing to give me any more information I literally can't do anything until i can find someone who has the same issue who can give me that information. I'm sorry....

**Comment by krogias** [comment]
What more info do you need? A log file?

Form my Android
Στις 22 Μαρ 2016 12:09 μ.μ., ο χρήστης "George Wilson" <
notifications@github.com> έγραψε:

**Comment by kigh143** [comment]
Sorry set it to yes
And make sure your using your email address and correct password

Sent from my ASUS

-------- Original Message --------

**Comment by roland-d** [comment]
@kigh143 @krogias  says he is not using a password to authenticate but he is being verified based on his IP address.

Unfortunately the error "Could not instantiate mail function" is very generic. So we will need to do some testing to see where it fails. I am going to look at where we can do some debugging to get this resolved.

**Comment by brianteeman** [comment]
Please see https://github.com/PHPMailer/PHPMailer/wiki/Troubleshooting#smtp-error-could-not-connect-to-smtp-host

You can enable phpmailer debugging which should help. This does sound like a server config issue but without the debug its hard to know

**Comment by roland-d** [comment]
You can change this file: https://github.com/joomla/joomla-cms/blob/staging/libraries/vendor/phpmailer/phpmailer/class.phpmailer.php#L326

for setting the debug mode.

The error is triggered at the moment that PHPMailer calls the PHP mail function. Interesting enough nothing has changed in this part of the PHPMailer code between the versions used in 3.4.8 and 3.5.0. If the debug mode doesn't show anything useful, you can try to remove the @ symbol before mail and see what error PHP throws at these lines:

https://github.com/joomla/joomla-cms/blob/staging/libraries/vendor/phpmailer/phpmailer/class.phpmailer.php#L675-L675

@krogias Are you using an SSL connection? If so, this may be happening in your case:

https://github.com/PHPMailer/PHPMailer/wiki/Troubleshooting#opportunistic-tls

Let me know what you find.

**Comment by krogias** [comment]
@roland-d No, I don't use ssl connection.

The NOTICE (not error) that I got was: **"SMTP connect() failed. https://github.com/PHPMailer/PHPMailer/wiki/Troubleshooting"**
I posted it here: http://forum.joomla.org/viewtopic.php?f=706&t=913705
I was a notice, but smtp stopped working.

@brianteeman Unfortunately I cannot reproduce the issue because I rolled back to 2.4.8. I cannot leave my live site without mailer.

**Comment by roland-d** [comment]
@krogias Would it be possible for you to set up just an empty test Joomla 3.5 site on your system for testing purposes? Without a system to test this on, it will be nearly impossible to trace the issue.

**Comment by krogias** [comment]
@roland-d I could do that. Thanks for the idea!

**Comment by roland-d** [comment]
@krogias Super, thanks. Let me know if you can find some more info.

**Comment by wilsonge** [comment]
@krogias any updates on this? Would be useful to find out whether it's that SSL issue as soon as possible so we can plan a course of attack on the bug here :)

**Comment by krogias** [comment]
Hi. I don't have access know. I will, tomorrow, from my office.
But I said that I don't and didn't use SSL.

**Comment by roland-d** [comment]
@umaha Do you have access to your system now to do some testing?

**Comment by umaha** [comment]
Yes, I do.

What should I do ?

Maha

On Tue, Mar 22, 2016 at 7:48 PM, RolandD <notifications@github.com> wrote:

> @umaha <https://github.com/umaha> Do you have access to your system now
> to do some testing?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/issues/9373#issuecomment-199835432>
>



-- 
R.Mahadevan
Manager - IT
248-504-6588 Ext. 6000 - Office
866-240-2160 - Toll Free
734-328-0234 -  Direct
www.advancedcashmanagement.com

This email contains information  that may be confidential, privileged,
proprietary or otherwise protected from disclosure. The email is intended
for the use only by the person(s) to whom it is addressed. If you are not
the intended recipient, any use, disclosure, copying, distribution,
printing or any action taken in reliance on the contents of this email, is
strictly prohibited. If you received this email in error, please contact
the sending party at the above address and delete the email. Thank you.

**Comment by umaha** [comment]
Yes, I have access now. WHat should I do?

**Comment by roland-d** [comment]
@umaha Please see my post https://github.com/joomla/joomla-cms/issues/9373#issuecomment-199778576 

I have put in some test instructions there. Thanks.

**Comment by umaha** [comment]
@roland-d 
referenced this issue 2 hours ago - Closed - send test email not working #9517

I used SMTP. I am not sure if that still uses the class.phpMailer.php. 

The emails from sites' contact form or some customhtml  work fine, ONLY the Send Test Email in Mail Settings does not.

I have only access and knowledge to work via the web interfaces.

**Comment by roland-d** [comment]
@umaha If possible can you email me Joomla and FTP access for that site so I can take a look to see what is happening? You can email me at roland.dalmulder@community.joomla.org

**Comment by umaha** [comment]
@roland-d 
OK , I can do that. I may need a few hours before I can start though.

**Comment by krogias** [comment]
@roland I have installed a test site. I could give you access if you like.

**Comment by roland-d** [comment]
@krogias, Yes please, if you can send me the details via email that would be appreciated.

**Comment by roland-d** [comment]
Thanks everybody, I am going to close this issue because we have a pull request #9528

@krogias Thank you very much for the test server. It has been valuable.

**Comment by AmaliDeveloper** [comment]
You can get the solution for send mail using phpmailer Here [Link](https://youtu.be/Edx7pPcPfs8)
